catalyst/callbacks/batch_transform.py

```diff
@@ -1,171 +1,162 @@
+                    dl.CriterionCallback(
+                        input_key="logits", target_key="targets", metric_key="loss"
+                    ),
+            # uncomment for extra metrics:
+            #       dl.AUCCallback(
+            #           input_key="scores", target_key="targets"
+            #       ),
+            #       dl.HitrateCallback(
+            #           input_key="scores", target_key="targets", topk_args=(1, 3, 5)
+            #       ),
+            #       dl.MRRCallback(
+            #           input_key="scores", target_key="targets", topk_args=(1, 3, 5)
+            #       ),
+            #       dl.MAPCallback(input_key="scores", target_key="targets", topk_args=(1, 3, 5)),
+            #       dl.NDCGCallback(
+            #           input_key="scores", target_key="targets", topk_args=(1, 3, 5)
+            #       ),
+                    dl.OptimizerCallback(metric_key="loss"),
+                    dl.SchedulerCallback(),
+                    dl.CheckpointCallback(
+                        logdir="./logs", loader_key="valid", metric_key="map01", minimize=False
+                    ),
+                ]
+            )
+
     """
 
     def __init__(
         self,
-        transform: Sequence[Union[dict, nn.Module]],
-        input_key: Union[str, int] = "image",
-        output_key: Optional[Union[str, int]] = None,
-    ) -> None:
-        """Init."""
-        super().__init__(order=CallbackOrder.Internal, node=CallbackNode.all)
+        transform: Callable,
+        scope: str,
+        input_key: Union[List[str], str] = None,
+        output_key: Union[List[str], str] = None,
+    ):
+        """
+        Preprocess your batch with specified function.
 
+        Args:
+            transform (Callable): Function to apply.
+            scope (str): ``"on_batch_end"`` or ``"on_batch_start"``
+            input_key (Union[List[str], str], optional): Keys in batch dict to apply function.
+            output_key (Union[List[str], str], optional): Keys for output.
+                If None then will apply function inplace to ``keys_to_apply``.
+                Defaults to ``None``.
+
+        Raises:
+            TypeError: When keys_to_apply is not str or list.
+        """
+        super().__init__(order=CallbackOrder.Internal)
+        if input_key is not None:
+            if not isinstance(input_key, (list, str)):
+                raise TypeError("input key should be stror list of str.")
+            elif isinstance(input_key, str):
+                input_key = [input_key]
+            self._handle_batch = self._handle_tuples
+        else:
+            self._handle_batch = self._handle_dicts
+
+        output_key = output_key or input_key
+        if output_key is not None:
+            if not isinstance(output_key, (list, str)):
+                raise TypeError("output key should be str or list of str.")
+            if isinstance(output_key, str):
+                output_key = [output_key]
```

Comment:
we also need to check, that `input_key is not None` in case of `output_key is not None`

Reply:
or it would lead to an exception